src/components/BookingModal/BookingForm/index.js

```diff
@@ -0,0 +1,168 @@
+import React, { Fragment } from 'react';
+import { PropTypes as PT } from 'prop-types';
+import container from './container';
+import { Form, Input } from '../../common';
+import eventTypes from '../../../utilities/eventTypes';
+import holidayStatus from '../../../utilities/holidayStatus';
+
+const BookingForm = props => {
+  const {
+    isEventBeingUpdated,
+    createEvent,
+    updateEvent,
+    formData,
+    formStatus,
+    formIsValid,
+    bookingDuration,
+    booking,
+    workingFromHomeBooking,
+    submitButtonDisabled,
+    availableDays,
+  } = props;
+
+  const {
+    eventStatus: { eventStatusId },
+  } = booking;
+  const isEventCancelled = eventStatusId === holidayStatus.CANCELLED;
+
+  const { eventTypeId } = formData;
+
+  const createCtas = () => {
+    const buttonTextValue = () => {
+      const durationText = bookingDuration === 0.5 ? 'Half' : bookingDuration;
+      const dayText = bookingDuration > 1 ? 'Days' : 'Day';
+
+      if (isEventBeingUpdated) {
+        if (isEventCancelled) {
+          return 'Cancelled';
+        }
+        return `Update to ${durationText} ${dayText}`;
+      } else {
+        if (eventTypeId !== eventTypes.ANNUAL_LEAVE) {
```

Comment:
should this be `if (eventTypeId === eventTypes.WFH)`? - it is possible that new event types will be added in future